src/models/CommonModel.ts

```diff
@@ -36,15 +36,39 @@ export class CommonModel extends CommonSchema<CommonModel> {
   /**
    * Set the types of the model
    * 
-   * @param types to set the model type to
+   * @param type
    */
-  setType(types : string | string[]) {
-    if (!Array.isArray(types) || types.length > 1) {
-      this.type = types;
-    } else if (types.length === 0) {
-      this.type = undefined;
-    } else if (types.length === 1) {
-      this.type = types[0];
+  setType(type : string | string[] | undefined) {
+    if (Array.isArray(type)) {
+      if (type.length === 0) {
+        this.type = undefined;
+        return;
+      } else if (type.length === 1) {
+        this.type = type[0];
+        return;
+      }
+    } 
+    this.type = type;
+  }
+  
+  /**
+   * Removes type(s) from model type
+   * 
+   * @param types 
+   */
+  removeType(typesToRemove : string | string[]) {
+    if (Array.isArray(typesToRemove)) {
+      for (const type of typesToRemove) {
+        this.removeType(type);
+      }
+    } else if (this.type !== undefined && this.type.includes(typesToRemove)) {
+      if (Array.isArray(this.type)) {
+        this.setType(this.type.filter((el) => {
+          return el !== typesToRemove;
+        }));
+      } else {
+        this.setType(undefined);
```

Comment:
You can here remove reassign types directly like `this.type = this.type.filter(el => el !== typesToRemove)` and `this.type = undefined` only for easier reading the code.

Reply:
`type` is just interpreted a little differently because it can be an array of a single value which is why I use the result of the filter and call `setType` so it handles that logic 😄